shor/quantum.py

```diff
@@ -43,6 +45,23 @@ def to_gates(self, include_operations=False) -> List[_Gate]:
             gates.extend(operation_or_gates)
         return gates
 
+    def to_registers(self):
+        qbit_registers = []
```

Comment:
We could use another Qbits class here instead of a list.
Same with Cbits.

One thing that does seem missing is the mapping between number and final index in qiskit circuit.

So:
qbits = Qbits(3)
qbits2 = Qbits(2)

I think we want to produce
`all_qbits_mapping = {0: (0, qbits), 1: (1, qbits), 2: (2, qbits), 3: (0, qbits2), 4: (1, qbits2) }`

Reply:
Let's chat about that

Reply:
Do you specifically want it to be a dictionary, the qbit and cbit classes are built for it to be a list of tuples?